design-proposals/emts-scale-provisioning.md

```diff
@@ -0,0 +1,321 @@
+- Have a UX to pre-register BareMetal edge nodes using Serial number or UUID or MAC address.
+- Provision different OS profiles to different edge nodes selected based on Serial number or UUID or MAC address.
+- Provision default OS when a device on the LAN boots over PXE and is not pre-registered.
+- Have a ux of  collecting provisioning logs and status of edge nodes.
+
+> Note: It might be possible for EMF-EIM to support provisioning of the EMT-S nodes. supporting this capability as part of MVP depends on any active customer requirements.
+
+## Solution
+
+The solution consists of two major enhancements (modifications) to EMF:
+1) **Support legacy PXE boot to scale EMT-S provisioning** - the EIM will be extended with a local DHCP/TFTP server that helps initiate OS provisioning via legacy PXE boot. 
+2) **Use EIM standalone deployment** - the proposed solution involves deploying a streamlined version of the EIM, referred to as EIM standalone,
+   at the customer's premises, specifically the OXM warehouse. This streamlined EIM will include only the essential components necessary for operating system provisioning. 
+   It is important to note that the EIM standalone will not be a separate branch or fork of the existing EIM system. 
+   The deployment strategy will utilize configuration settings at deployment time within the EMF to ensure that only the required features are activated.
+   The detailed design for the deployment time configuration and profile settings for both EMF and EIM will be addressed in a separate design document.
+
+### Legacy PXE provisioning workflow
+
+The solution relies on the capabilities provided by Tinkerbell SMEE. SMEE provides a minimal DHCP and TFTP server implementations that support PXE boot.
+The DHCP server provides necessary PXE info to machines initiating the PXE boot, while the TFTP server provides the iPXE binary that can be chain-loaded at boot time by PXE.
+The iPXE binary provided by the SMEE's TFTP server is a third-party script, maintained by the Tinkerbell project.
+
+The solution assumes that the local EIM-provided DHCP/TFTP server is only used to initiate boot via legacy PXE (as legacy PXE has limited capabilities, i.e., it doesn't support HTTPS boot),
+but, once chain-loaded to iPXE, the Micro-OS is downloaded and used to drive provisioning.
+In other words, the only difference between the new PXE-based boot and HTTP-based boot is how the OS provisioning is triggered. The subsequent workflow remains the same -
+it leverages Micro-OS, device discovery and Tinkerbell workflow to complete OS provisioning.
+
+**NOTE1**: Customers should provide their own local DHCP server for dynamic IP address assignment.
+
+**NOTE2**: The workflow assumes that the EIM Standalone is deployed locally on customers' premises and ENs have direct connectivity with EIM services.
+
+The high-level PXE-based provisioning workflow is as follows:
```

Comment:
also in the sequence say something about external DHCP interactions. Or create a small paragraph with the assumptions and the checklist user should comply:

- Boot works in proxy dhcp mode only
- External DHCP server uses DHCP options....

Reply:
done